src/app/api/server.ts

```diff
@@ -430,94 +371,45 @@ export async function initializeApi(agent: SaikiAgent, agentCardOverride?: Parti
+                    const currentConfig = agent.getEffectiveConfig(sessionId);
+                    const validation = validateInputForLLM(
+                        {
+                            text: data.content,
+                            ...(imageDataInput && { imageData: imageDataInput }),
+                            ...(fileDataInput && { fileData: fileDataInput }),
+                        },
+                        {
+                            provider: currentConfig.llm.provider,
+                            model: currentConfig.llm.model,
                         }
+                    );
 
-                        // Model-specific file format validation
-                        const sessionId = data.sessionId as string | undefined;
-                        const currentConfig = agent.getEffectiveConfig(sessionId);
-                        const { validateFileForLLM, createFileValidationError } = await import(
-                            '@core/ai/llm/validation.js'
-                        );
+                    if (!validation.isValid) {
+                        const errorDetails = createInputValidationError(validation, {
+                            provider: currentConfig.llm.provider,
+                            model: currentConfig.llm.model,
+                        });
 
-                        const validation = validateFileForLLM(
-                            {
-                                provider: currentConfig.llm.provider,
-                                model: currentConfig.llm.model,
-                            },
-                            mimeType
+                        ws.send(
+                            JSON.stringify({
+                                event: 'error',
+                                data: errorDetails,
+                            })
                         );
-
-                        if (!validation.isSupported) {
-                            const errorDetails = createFileValidationError(validation, {
-                                provider: currentConfig.llm.provider,
-                                model: currentConfig.llm.model,
-                            });
-
-                            ws.send(
-                                JSON.stringify({
-                                    event: 'error',
-                                    data: errorDetails,
-                                })
-                            );
-                            return;
-                        }
-
-                        fileDataInput = { data: base64, mimeType, filename };
+                        return;
                     }
 
-                    const sessionId = data.sessionId as string | undefined;
                     const stream = data.stream === true; // Extract stream preference, default to false
                     if (imageDataInput) logger.info('Image data included in message.');
```

Comment:
nit: move logs in line 414-416 before validation, similar in other apis also
flow would then be -> log the inputs, validate the inputs, return the response

currently if validation fails earlier we wouldn't see logs which could be harder to debug